### PR TITLE
Add FooterActionButton to MarkdownEditor

### PR DESCRIPTION
**Pull request by JelloBagel**
Closes https://github.com/github/primer/issues/2174

I added the ability to add custom footer buttons before the "Markdown is Supported" in the MarkdownEditor. I created `MarkdownEditor.FooterButton` and `MarkdownEditor.Footer` components to align with the approach taken for `MarkdownEditor.Toolbar` and `MarkdownEditor.ToolbarButton`.

### Screenshots

Before
image

After
image

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

**Comment by lesliecdubs** [comment]
Hi @JelloBagel 👋🏻 Thanks for the contribution! Are you still working on this? If you're ready for a review, please bring this out of "draft" mode and a reviewer will be auto-assigned.

File: docs/content/drafts/MarkdownEditor.mdx [hunk 1]
```diff
@@ -28,9 +28,8 @@ import {MarkdownEditor} from '@primer/react/drafts'
 
 A `Label` is always required for accessibility:
 
-{/* prettier-ignore-start */}
```
Comment (JelloBagel) [inline docs/content/drafts/MarkdownEditor.mdx:31]: @josepmartins I'm not seeing the these mdx changes I made here in the [github-pages deployment](https://primer-cf028944c7-13348165.drafts.github.io/drafts/MarkdownEditor). Do you know why the changes would not be reflected? 

Also, let me know if there is any additional steps to move this PR forward.

**Comment by mperrotti** [comment]
@JelloBagel - the deployment issue is very strange. I haven't seen that before. Can you confirm that it looks as you'd expect when you build locally?

**Comment by JelloBagel** [comment]
@mperrotti I ran `build:docs:preview` (was this the right command?) and can confirm that my changes are reflected in the build

Before           |  After
:-------------------------:|:-------------------------:
Screenshot 2023-06-09 at 11 39 17 AM |  Screenshot 2023-06-09 at 11 39 25 AM

File: src/drafts/MarkdownEditor/MarkdownEditor.test.tsx [hunk 2]
```diff
@@ -57,6 +57,8 @@ const render = async (ui: React.ReactElement) => {
 
   const queryForToolbarButton = (label: string) => within(getToolbar()).queryByRole('button', {name: label})
 
+  const getDefaultFooterButton = () => within(getFooter()).getByRole('link', {name: 'Markdown documentation'})
```
Comment (marywhite) [inline src/drafts/MarkdownEditor/MarkdownEditor.test.tsx:60]: minor suggestion, we could add some test coverage here to assert that default footer button appears when not using a custom footer

File: src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx [hunk 3]
```diff
@@ -288,6 +293,113 @@ export const CustomButtons = ({
           <MarkdownEditor.ToolbarButton icon={DiffIcon} onClick={onDiffClick} aria-label="Custom Button" />
           <MarkdownEditor.DefaultToolbarButtons />
         </MarkdownEditor.Toolbar>
+      </MarkdownEditor>
+      <p>Note: for demo purposes, files starting with &quot;A&quot; will be rejected.</p>
+    </>
+  )
+}
+
+export const CustomFooter = ({
```
Comment (marywhite) [inline src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx:302]: Not required, but initially I had to dig to understand the differences between `CustomFooter` and `CustomFooterActions` given that they both use the action buttons. What do you think about consolidating, or alternatively removing the actions here?

**Review by marywhite — COMMENTED** [review]
I am not a primer reviewer, so I'll abstain from any official review. This looks great to me, however! I left a few minor suggestions, but consider them optional.

File: src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx (hunk 3 above)
Comment (JelloBagel) [reply src/drafts/MarkdownEditor/MarkdownEditor.stories.tsx:302]: I left `CustomFooterActions` story to support the existing implementations with the `MarkdownEditor` that use the custom actions, but I would like to consolidate it to use the `CustomFooter` once its published.

**Review by josepmartins — APPROVED** [review]
